lib/buildkite/test_collector/minitest_plugin/trace.rb

```diff
@@ -72,17 +70,20 @@ def project_dir
     end
 
     def failure_reason
-      @failure_reason ||= example.failure&.message
+      @failure_reason ||= strip_invalid_utf8_chars(example.failure&.message)&.split("\n")&.first
     end
 
     def failure_expanded
-      @failure_expanded ||= begin
-        example.failures.map do |failure|
-          {
-            expanded: failure.message,
-            backtrace: failure.backtrace,
-          }
-        end
+      @failure_expanded ||= example.failures.map.with_index do |failure, index|
+        # remove the first line of message from the first failure
+        # to avoid duplicate line in Test Analytics UI
+        messages = strip_invalid_utf8_chars(failure.message).split("\n")
+        messages = messages[1..] if index.zero?
```

Comment:
I don't understand why we're doing this only for the first failure? Shouldn't we be doing this for all failures? 

I don't know heaps about minitest - but it appears that there could be multiple failures here, much like how rspec does aggregate failures.